File: Nasa_images/Nasa_imagesTests/Nasa_imagesTests.m
//
//  Nasa_imagesTests.m
//  Nasa_imagesTests
//
//  Created by CHRISTIAN BEYNIS on 8/6/22.
//

#import <XCTest/XCTest.h>
#import "NetworkManager.h"



@interface Nasa_imagesTests : XCTestCase

@property NetworkManager *NtwkTest;
@property (nonatomic, strong) NSMutableArray* NasaInfos;


@end

@implementation Nasa_imagesTests

- (void)setUp {
    // Put setup code here. This method is called before the invocation of each test method in the class.
    _NtwkTest = [[NetworkManager alloc] init];
    self.NasaInfos = [[NSMutableArray alloc] init];
    
   
    
}

- (void)tearDown {
    // Put teardown code here. This method is called after the invocation of each test method in the class.
}

- (void)testfetchNasaInfo {
//    _NasaInfos = [];
    [[NetworkManager sharedInstance] fetchNasaInfo: ^(NSMutableArray * _Nullable NasaInfo) {
        [self.NasaInfos addObjectsFromArray: NasaInfo];
        //self.NasaInfos.count == NasaInfo.count;
        NSLog(@"ok");
        XCTAssertTrue(self.NasaInfos.count == NasaInfo.count);
        XCTAssertEqual(self.NasaInfos.count, 1);
        
        
        
        
      }] ;
    // This is an example of a functional test case.
    // Use XCTAssert and related functions to verify your tests produce the correct results.
    
    //XCTAssertEqual(self.NasaInfos.count,1);

    XCTAssertTrue(self.NasaInfos.count == 0);
    
}

- (void)testPerformanceExample {
    // This is an example of a performance test case.
    [self measureBlock:^{
        // Put the code you want to measure the time of here.
    }];
}

@end

File: Nasa_images/Nasa_imagesTests/Nasa_imagesTests.swift
//
//  Nasa_imagesTests.swift
//  Nasa_imagesTests
//
//  Created by CHRISTIAN BEYNIS on 8/8/22.
//

import XCTest
@testable import Nasa_images

class NetworkManagerTests: XCTestCase {

    var networkManager: NetworkManager?
    
    private var NasaInfos: [NasaImg] = []
    
    override func setUpWithError() throws {
        try super.setUpWithError()
        self.networkManager = NetworkManager.sharedInstance()
        
    }

    override func tearDownWithError() throws {
        self.networkManager = nil
        try super.tearDownWithError()
    }

    func testGetModelDataSuccess() {
        // Arrange  // models = NasaInfos
        
        var NasaInfosTest: [NasaImg] = []
        
//        let expectation = XCTestExpectation(description: "Successfully Fetched Movies")
        
        // Act
        
        
        self.networkManager?.fetchNasaInfo{  NasaInfo in
            
                guard let NasaInfos = NasaInfo as? [NasaImg]
                else {
                    print("Nothing")
                    return
                    
                }
                self.NasaInfos.append(contentsOf: NasaInfos)
//                expectation.fulfill()
                NasaInfosTest = self.NasaInfos
            }
                XCTAssert(NasaInfosTest.count == 0)
        
        }
        
        
//        self.networkManager?.fetchNasaInfo {
//            [weak self] NasaInfo in
//
//                guard let NasaInfos = NasaInfo as? [NasaImg] else { return }
//                self?.NasaInfos.append(contentsOf: NasaInfos)
//
//            }
//
//
//        // Assert
//        XCTAssert(self.NasaInfos.count==0)
//        //XCTAssertEqual(NasaInfos.first?.title, "The Full Moon of 2021")
//
//    }
    
    

}

